### ISSUE TITLE
Cannot restore database with asymmetric encryption

### ISSUE THREAD

**Issue by ninjalf2** [duplicati/duplicati#3533]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 0.0.0.0
- **Operating system**: Windows 10 64-bit
- **Backend**: SFTP (SSH)

## Description

My backups are encrypted using asymmetric encryption; I followed [this](https://github.com/duplicati/duplicati/issues/2270) to set it up. It works fine for backing up and restoring. I've recently switched to using Duplicati as a service which broke my database, because the database is now located in a different place. I guess I could just figure out where the database files are located and move them to the correct place, but I still want to be able to restore in the future anyway.

When I try to restore I get a bunch of errors with "Zip archive appears to have a broken Central Record Header, switching to stream mode". They are most likely not broken since my backups haven't been interrupted. My guess is that it cannot decrypt it properly or something.

I am not sure if this has anything to do with running it as a service but I'm guessing and hoping that it doesn't. I'm also a bit reluctant to test it out by restoring the database with Duplicati running as a normal user as I will then have to backup hundreds of GB again in case it doesn't work.

## Steps to reproduce
1. Backup using asymmetric encryption [(follow this)](https://github.com/duplicati/duplicati/issues/2270l)
2. Restore database

## Debug log
[bugreport.zip](https://github.com/duplicati/duplicati/files/2660738/bugreport.zip)

**Comment by warwickmm** [comment]
It appears that the issues lie with the following files:
* `duplicati-20181129T164728Z.dlist.zip.gpg`
* `duplicati-20181129T182507Z.dlist.zip.gpg`

Are you able to download these and decrypt and unzip them?

**Comment by ninjalf2** [comment]
Both of the files decrypt and unzip successfully. They contain two JSON files, filelist.json and manifest.

Is there any other information I can provide to help narrow down the issue?

**Comment by warwickmm** [comment]
In previous cases, this error was an indication that the zip files were damaged.  I'm not sure what may be causing this, so we may have to see if others can chime in to provide additional insight.

**Comment by ninjalf2** [comment]
@warwickmm Am I right to assume that Duplicati would need to decrypt the files to be able to restore the database? If yes, then my theory may be correct since Duplicati does not ask for a password when trying to restore the database like it does when it needs to restore files.

**Comment by Pectojin** [comment]
Yes, all dlist files must be downloaded and decrypted to recreate the database.

**Comment by vh-discuss** [comment]
I had the same problem when I tried to repair the database via web interface. Then I used the CLI
- Recreate the configuration (asymmetric encryption, passphrase "unused"
- using the cli to repair: Here I was asked for the password. where in fact I had to provide the password of my private key
- after that downloading and decrypting of the dlist-files worked like a charm
BUT:
- next time I use the web interface it complains that I tried to change the passphrase and refuses any action :-(

From my point of view it would be nice to implement what ninjalf2 suggested. Providing the password of the secret key interactive but don't save it (same action when restoring files). Maybe there are other hooks where we need it, too: Checking files, compressing backup.
